nbqa/__main__.py

```diff
@@ -323,8 +323,13 @@ def _get_configs(cli_args: CLIArgs, project_root: Path) -> Configs:
     # If a section was passed via CLI, use that.
     for section in config:
         if getattr(cli_args, section) is not None:
-            # TypedDict key must be a string literal
-            config[section] = getattr(cli_args, section)  # type: ignore
+            if section == "addopts":
+                # addopts are added to / overridden rather than replaced outright
+                # Sequence[str] has no attribute extend. Should this be typed as List[str]?
+                config["addopts"].extend(getattr(cli_args, section))  # type: ignore
```

Comment:
As noted in the comment, because `config["addopts"]` is typed as a `Sequence[str]`, Mypy doesn't like the usage of `extend`. Should it be typed as `List[str]` instead? I haven't checked yet if it could ever be anything other than a list.

Reply:
Having taken a quick look, it seems the default is an empty list, and will only be something else if overwritten by the parsed `pyproject.toml`.

I think the only way there could be a problem is if the `pyproject.toml` itself is misspecified, e.g. if someone did something like

```toml
[tool.nbqa.addopts]
black = "invalid"
```

In that case you probably have bigger problems, so perhaps the more explicit typing would be ok? Please tell me if there's something I missed though.

Reply:
Would something like
```
config["addopts"] = (*config["addopts"], *getattr(cli_args, section))
```
so that we don't lose immutability in the rest of the codebase?

Reply:
Sure, have changed as you suggested. 

To clarify though the value of `config["addopts"]` was previously a (mutable) list, are you referring to not using in-place operations such as `.extend` rather than avoiding mutable data structures?